### PR TITLE
Bookmark load/save

### PR DESCRIPTION
**Pull request by PatrikLundgren**
Bookmarks are now stored in projects and saved upon saving the  project. When project is loaded, the bookmarks are loaded back into ViAn. Projects have been extended to contain VideoProjects, which contain s a video and its associated bookmarks.

Bookmarks need to have more sophisticated handling as the current implementation suffers from some problems:
1: Very messy doc.list when many projects are loaded (or just one with many bookmarks).
2: When jumping to a bookmark, it seems to jump to a frame behind the frame it was saved as.
3: Bookmarks from all videos in a project is saved in a single subfolder of that project. Makes for a messy folder.
4: If 2 bookmarks are saved with the same frame number. The second will overwrite the first. This is because 
bookmarks are saved with a filepath dependant on the frame of the bookmark.

However, these problem i consider to be problems for further implementation. This extension is just extending previous behaviour of bookmarks to be loaded and saved.

### CODE REVIEW

File: ViAn/GUI/mainwindow.cpp [hunk 1]
```diff
@@ -335,17 +336,23 @@ void MainWindow::on_bookmarkButton_clicked() {
...
 
-        bookmark_view->add_bookmark(mvideo_player->get_current_frame_num(), file_path);
+        std::string file_name = std::to_string(bookmark_view->get_num_bookmarks());
+        // Get bookmark description
+        QString bookmark_text("");
+        bool ok;
+        bookmark_text = bookmark_view->get_input_text(&ok);
+        if(!ok) return;
+
+        std::string file_path = mvideo_player->export_current_frame(dir.absolutePath().toStdString(), std::to_string(mvideo_player->get_current_frame_num()));
```
Comment (GeorgeYildiz) [inline ViAn/GUI/mainwindow.cpp]: Maybe just make local variables to save the arguments so this line dosent become so long. I think its good to keep a code line below 80 symbols.

File: ViAn/Filehandler/videoproject.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    return this->video;
+}
+
+/**
+ * @brief VideoProject::get_bookmarks
+ * @return bookmarks
+ * Return all bookmarks.
+ */
+std::vector<Bookmark *> VideoProject::get_bookmarks()
+{
```
Comment (parso619) [inline ViAn/Filehandler/videoproject.cpp]: `{` on new line

File: ViAn/Filehandler/project.cpp [hunk 3]
```diff
@@ -54,23 +56,44 @@ void Project::remove_video(ID id){
...
+    vid_proj->get_video()->id = this->v_id;
+    this->videos.insert(std::make_pair(this->v_id, vid_proj));
+    return this->v_id++;
+}
+/**
+ * @brief Project::delete_artifacts
+ * Delete all projects files.
+ */
+void Project::delete_artifacts()
+{
```
Comment (parso619) [inline ViAn/Filehandler/project.cpp]: `{` on new line

File: ViAn/ViAn.pro [hunk 4]
```diff
@@ -15,7 +15,9 @@ TEMPLATE = app
 #
 # GENERAL
 #
-SOURCES += main.cpp
+SOURCES += main.cpp \
+    Filehandler/videoproject.cpp \
+    Filehandler/bookmark.cpp
```
Comment (parso619) [inline ViAn/ViAn.pro]: You've forgotten to put the sources and headers in the correct places n the pro-file.

File: ViAn/GUI/mainwindow.cpp [hunk 5]
```diff
@@ -336,17 +337,25 @@ void MainWindow::on_bookmark_button_clicked() {
...
-        QDir proj_path = fileHandler->get_dir(my_project->id);
         // Add bookmarks-folder to the project-folder.
-        proj_path.mkpath(proj_path.absoluteFilePath("Bookmarks"));
-        proj_path.cd(proj_path.absoluteFilePath("Bookmarks"));
-
+        Project* proj = fileHandler->get_project(my_project->id);
+        QDir dir = fileHandler->get_dir(proj->bookmark_dir);
         // Export the current frame in the bookmarks-folder.
         // The names of the stored files will have increasing numbers.
+
```
Comment (parso619) [inline ViAn/GUI/mainwindow.cpp]: This comment is connected to the code a couple lines below:
`std::string file_name = std::to_string(bookmark_view->get_num_bookmarks());`

File: ViAn/GUI/bookmarkitem.h [hunk 6]
```diff
@@ -1,17 +1,17 @@
 #ifndef BOOKMARKITEM_H
 #define BOOKMARKITEM_H
 
-#include "bookmark.h"
+#include "../Filehandler/bookmark.h"
```
Comment (parso619) [inline ViAn/GUI/bookmarkitem.h]: I don't believe `../` is necessary.

**Pull request merged**